utilities/tree.metta

```diff
@@ -30,6 +30,15 @@
             (mkTree (mkNode $expr) Nil))
             )
 
+(: cleanTree (-> (Tree $a) (Tree $a)))
+(= (cleanTree (mkTree (mkNode $r) (Cons $x $xs)))
```

Comment:
You don't need to deconstruct the tree here, you could just use a variable as `$tree` and use that in the preOrder function. That would reduce the complexity of the function.